### ISSUE TITLE
Not working with ipywidgets_bokeh and Panel

### ISSUE THREAD

**Issue by MarcSkovMadsen** [martinRenou/ipycanvas#236]
CROSS POST: https://github.com/bokeh/ipywidgets_bokeh/issues/44

Panel: 0.12.6 Bokeh: 2.4.2 ipywidgets_bokeh: 1.2.1 ipycanvas: 0.10.2 ipykernel: 6.6.0

Raises `expected a string or ArrayBuffer, got object`.

![image](https://user-images.githubusercontent.com/42288570/146634010-8a565cf4-1d64-4c1a-8409-2464b5730d9b.png)

```python
# pip install ipycanvas panel ipywidgets_bokeh
import panel as pn
from ipycanvas import Canvas

pn.extension('ipywidgets')

canvas = Canvas(width=200, height=200)

# Cubic curves example
def draw():
    canvas.begin_path()
    canvas.move_to(75, 40)
    canvas.bezier_curve_to(75, 37, 70, 25, 50, 25)
    canvas.bezier_curve_to(20, 25, 20, 62.5, 20, 62.5)
    canvas.bezier_curve_to(20, 80, 40, 102, 75, 120)
    canvas.bezier_curve_to(110, 102, 130, 80, 130, 62.5)
    canvas.bezier_curve_to(130, 62.5, 130, 25, 100, 25)
    canvas.bezier_curve_to(85, 25, 75, 37, 75, 40)
    canvas.fill()
    print("drawn")

pn.state.onload(draw)

pn.panel(canvas, height=200, width=200).servable()
```

**Comment by MarcSkovMadsen** [comment]
## Additional Context

Panel should work with ipywidgets via `ipywidgets_bokeh`. See https://panel.holoviz.org/reference/panes/IPyWidget.html#

**Comment by martinRenou** [comment]
It might be an issue with Bokeh itself. I see the error raises in `ipywidgets_bokeh.js`, which is not ipycanvas's JavaScript code nor ipywidgets's one.

**Comment by MarcSkovMadsen** [comment]
We had a similar issue with ipyleaflet. We investigated `panel`, `ipywidgets_bokeh` and `ipyleaflet`. It turned out it was an error in `ipyleaflet` https://github.com/jupyter-widgets/ipyleaflet/pull/873. So it can be anywhere. You where actually involved in solving that one.  😄 Thanks.

**Comment by martinRenou** [comment]
Ok. Does the error show up also when not trying to draw anything? Are you able to show an empty canvas?

**Comment by MarcSkovMadsen** [comment]
Yes

```python
# pip install ipycanvas panel ipywidgets_bokeh
import panel as pn
from ipycanvas import Canvas

pn.extension('ipywidgets')

canvas = Canvas(width=200, height=200)

pn.panel(canvas, height=200, width=200).servable()
```

![image](https://user-images.githubusercontent.com/42288570/146746601-268deec0-b16e-4b09-b8d0-9f0d402a0b7c.png)

![image](https://user-images.githubusercontent.com/42288570/146746684-5ec8622e-85d4-49c3-9944-7116b44591cf.png)

**Comment by MarcSkovMadsen** [comment]
With Ipyleaflet the below had to be added.

![image](https://user-images.githubusercontent.com/42288570/146746909-14a5e115-e686-440e-a4f1-095e3c172bc7.png)

Could that be the problem here as well?

**Comment by martinRenou** [comment]
I don't think so. If the Canvas shows up, then the canvas widget model is less likely to be the problem.

I think it's an issue with binary buffers in custom comm messages in bokeh.

Can you click here? 
![error](https://user-images.githubusercontent.com/21197331/146747333-13d9a37e-fe42-49b3-a6d3-751130f29fdf.png)
So we can see in which part of the code it fails?

**Comment by MarcSkovMadsen** [comment]
![image](https://user-images.githubusercontent.com/42288570/146748012-8a93a508-1ad9-4ad4-a15d-d48681582ded.png)

**Comment by martinRenou** [comment]
Would you be able to open the debugger at this point and show what `e` contains?

**Comment by MarcSkovMadsen** [comment]
This is an example. I don't believe this is the one failing.

![image](https://user-images.githubusercontent.com/42288570/146748713-9b172eee-7aad-44f0-a2d1-ec024710f73f.png)

I will try to debug and share.

**Comment by MarcSkovMadsen** [comment]
It seems its when `e` is `null`

![image](https://user-images.githubusercontent.com/42288570/146749291-17388524-a045-4bcf-9850-247eb7499963.png)

**Comment by martinRenou** [comment]
I don't know what this `ipywidgets_bokeh` message is. Is it some custom messaging logic that Bokeh uses for wrapping all Jupyter messages?

**Comment by martinRenou** [comment]
It might be worth looking what happens Python side with this `ipywidges_bokeh`, there might be a reason why the message is `null` (or `None` Python side?)

**Comment by MarcSkovMadsen** [comment]
Probably. But I am not familiar with `ipywidgets_bokeh` either.

But the source js line is here https://github.com/bokeh/ipywidgets_bokeh/blob/379308211dd81e67eeb792dab2c58d5e98598cc2/ipywidgets_bokeh/src/widget.ts#L73

**Comment by MarcSkovMadsen** [comment]
Refactoring my example to

```python
# pip install ipycanvas panel ipywidgets_bokeh
import panel as pn
from ipycanvas import Canvas

pn.extension('ipywidgets')

canvas = Canvas(width=200, height=200)

# Cubic curves example
button = pn.widgets.Button(name="run")
@pn.depends(button.param.clicks, watch=True)
def draw(clicks):
    print("draw begin")
    canvas.begin_path()
    canvas.move_to(75, 40)
    canvas.bezier_curve_to(75, 37, 70, 25, 50, 25)
    canvas.fill()
    print("draw end")

pn.Column(
    button,
    pn.panel(canvas, height=200, width=200)
).servable()
```

I can see that the message is not send from that function as it does not print anything if I run the `draw` function. But the error on client side appears.

https://user-images.githubusercontent.com/42288570/146754367-77e84009-9dff-4954-b2bc-a16b3adee67e.mp4
